pkg/operator/resourcesynccontroller/resourcesync_controller.go

```diff
@@ -171,13 +185,13 @@ func (c *ResourceSyncController) Sync(ctx context.Context, syncCtx factory.SyncC
 			continue
 		}
 
-		_, _, err := resourceapply.SyncConfigMap(c.configMapGetter, syncCtx.Recorder(), source.Namespace, source.Name, destination.Namespace, destination.Name, []metav1.OwnerReference{})
+		_, _, err := resourceapply.SyncPartialConfigMap(c.configMapGetter, syncCtx.Recorder(), source.Namespace, source.Name, destination.Namespace, destination.Name, source.SyncedKeys, []metav1.OwnerReference{})
```

Comment:
seeing partial synchronization inside `Sync` method immediately yields a warning sign. Do you think it would make sense to split the code so that we could have two code paths? One for `SyncPartialConfigMap` and the other one for `SyncConfigMap` ?

Reply:
> seeing partial synchronization inside Sync method immediately yields a warning sign
I don't understand, partial is just a special case of full

Why split into two separate code paths when the code works exactly the same? We already have a separate code path for secrets and configmaps, even though the actions are more or less equal, why add two other copy-pasted code paths?

I could of course do 
```golang
if len(source.SyncedKeys) > 0 {
    _, _, err := resourceapply.SyncPartialConfigMap(c.configMapGetter, syncCtx.Recorder(), source.Namespace, source.Name, destination.Namespace, destination.Name, source.SyncedKeys, []metav1.OwnerReference{})
} else {
    resourceapply.SyncConfigMap(c.configMapGetter, syncCtx.Recorder(), source.Namespace, source.Name, destination.Namespace, destination.Name, []metav1.OwnerReference{})
}
```
but `resourceapply.SyncConfigMap()` resolves directly to `esourceapply.SyncPartialConfigMap()` with `nil` `syncedKeys`